libs/platform/src/lib/components/form/datetime-picker/datetime-picker.component.ts

```diff
@@ -0,0 +1,345 @@
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    EventEmitter,
+    Host,
+    Input,
+    OnDestroy,
+    OnInit,
+    Optional,
+    Output,
+    Self,
+    SkipSelf,
+    ViewChild
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import {
+    CalendarYearGrid,
+    DatetimePickerComponent,
+    DaysOfWeek,
+    FdCalendarView,
+    FdDate,
+    FdDatetime,
+    SpecialDayRule
+} from '@fundamental-ngx/core';
+import { Placement } from 'popper.js';
+import { BaseInput } from '../base.input';
+import { FormField } from '../form-field';
+import { FormFieldControl, Status } from '../public_api';
+
+@Component({
+    selector: 'fdp-datetime-picker',
+    templateUrl: './datetime-picker.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [{ provide: FormFieldControl, useExisting: PlatformDatetimePickerComponent, multi: true }]
+})
+export class PlatformDatetimePickerComponent extends BaseInput implements OnInit, AfterViewInit, OnDestroy {
+    /**
+     * value for datetime in string
+     */
+    @Input()
+    get value(): string {
+        return super.getValue();
+    }
+    set value(value: string) {
+        super.setValue(value);
+    }
```

Comment:
Is it really should be a string?
I assumed we should treat the `FdDatetime` instance...

Reply:
this is because core datetimepicker also accepts string as the input and internally converts it to an `FdDatetime` instance if it is a valid datetime. Initially when I had `FdDatetime` I also remember that on key press changes, it would not reflect the keyed in changes but rather reset the date(something the core dtp functionality does internally). To keep things simpler, I am accepting string and letting core dtp do the other manipulations internally.

Reply:
Neither `date` input nor `writeValue` interface are expecting a string in the core version:
https://github.com/SAP/fundamental-ngx/blob/71200264b59bf4828f0c1a4fa32c05aae65390a5/libs/core/src/lib/datetime-picker/datetime-picker.component.ts#L128-L130
https://github.com/SAP/fundamental-ngx/blob/71200264b59bf4828f0c1a4fa32c05aae65390a5/libs/core/src/lib/datetime-picker/datetime-picker.component.ts#L409

Reply:
Could you please pointme where the core dtp is referring to string in its inputs?

Reply:
https://github.com/SAP/fundamental-ngx/blob/71200264b59bf4828f0c1a4fa32c05aae65390a5/libs/core/src/lib/datetime-picker/datetime-picker.component.ts#L488-L493

used here:
https://github.com/SAP/fundamental-ngx/blob/71200264b59bf4828f0c1a4fa32c05aae65390a5/libs/core/src/lib/datetime-picker/datetime-picker.component.html#L13-L23

Reply:
`handleInputChange` is not an input, that's how component reacts on entering a date as a string via keyboard.
`@input value` is not about this

Reply:
forgot to update here, but this is fixed now; setting FdDatetime object for value.